x/bank/legacy/v040/keys.go

```diff
@@ -0,0 +1,47 @@
+package v040
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	v040auth "github.com/cosmos/cosmos-sdk/x/auth/legacy/v040"
+)
+
+const (
+	// ModuleName defines the module name
+	ModuleName = "bank"
+
+	// StoreKey defines the primary module store key
+	StoreKey = ModuleName
+
+	// RouterKey defines the module's message routing key
+	RouterKey = ModuleName
+
+	// QuerierRoute defines the module's query routing key
+	QuerierRoute = ModuleName
+)
+
+// KVStore keys
+var (
+	BalancesPrefix      = []byte("balances")
+	SupplyKey           = []byte{0x00}
+	DenomMetadataPrefix = []byte{0x1}
+)
+
+// DenomMetadataKey returns the denomination metadata key.
+func DenomMetadataKey(denom string) []byte {
+	d := []byte(denom)
+	return append(DenomMetadataPrefix, d...)
+}
+
+// AddressFromBalancesStore returns an account address from a balances prefix
+// store. The key must not contain the perfix BalancesPrefix as the prefix store
+// iterator discards the actual prefix.
+func AddressFromBalancesStore(key []byte) sdk.AccAddress {
+	addr := key[:v040auth.AddrLen]
```

Comment:
These 2 checks are unnecessary and the first one is bound to panic if I pass in a key less than v040auth.AddrLen. Please remove this one, or move if after ensure that there are at least v040auth.AddrLen values.

Reply:
All this code (in the `legacy/` folder) is copy-pasted from a previous commit of the SDK. To avoid breaking the migration scripts inadvertently, I prefer to keep them as is.

Accessing key slices before checking key length also makes me feel uncomfortable, I created https://github.com/cosmos/cosmos-sdk/issues/8451 for this.

Reply:
Thank you @amaurymartiny!